test/test_collections.py

```diff
@@ -339,5 +339,34 @@ def test_from_httplib_py2(self):
         self.assertEqual(d['www-authenticate'], 'asdf, bla')
         self.assertEqual(d.getlist('www-authenticate'), ['asdf', 'bla'])
 
+    def test_from_httplib_py2_invalid_first_header(self):
+        if six.PY3:
+            raise SkipTest("python3 has a different internal header implementation")
```

Comment:
If we're implementing something that replicates Python 3's behaviour, shouldn't this test behave identically on Python 2 and Python 3?